flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/ContaineredTaskManagerParameters.java

```diff
@@ -90,6 +93,12 @@ public static ContaineredTaskManagerParameters create(
             }
         }
 
+        // set JAVA_HOME
+        String javaHome = config.get(CoreOptions.FLINK_JAVA_HOME);
+        if (!javaHome.isEmpty()) {
```

Comment:
I think `config.get(CoreOptions.FLINK_JAVA_HOME)` will return `""` if the `env.java.home` is not configured. 

Can you add a negative test case to validate that the default Java home is picked when `CoreOptions.FLINK_JAVA_HOME` is not set?

Reply:
When `CoreOptions.FLINK_JAVA_HOME` has no default value, `config.get(CoreOptions.FLINK_JAVA_HOME)` will return `null` if `env.java.home` is not configured. I'll update the usages to accommodate the default value change, and add UTs on it. Since we cannot directly test if the default Java home is picked in UT, I'll test it in the reverse way: if the env `JAVA_HOME` is not overridden by Flink, then we are good